extensions/cast/src/main/java/com/google/android/exoplayer2/ext/cast/CastPlayer.java

```diff
@@ -145,42 +145,57 @@ public CastPlayer(CastContext castContext) {
     pendingSeekWindowIndex = C.INDEX_UNSET;
     pendingSeekPositionMs = C.TIME_UNSET;
     updateInternalState();
+
+    if (remoteMediaClient != null) {
+      try {
+        remoteMediaClient.addListener(statusListener);
+        remoteMediaClient.addProgressListener(statusListener, PROGRESS_REPORT_PERIOD_MS);
+        updateInternalState();
+      } catch (NullPointerException ignored) {
+      }
+    }
   }
 
   // Media Queue manipulation methods.
 
   /**
    * Loads a single item media queue. If no session is available, does nothing.
    *
-   * @param item The item to load.
+   * @param item       The item to load.
    * @param positionMs The position at which the playback should start in milliseconds relative to
-   *     the start of the item at {@code startIndex}. If {@link C#TIME_UNSET} is passed, playback
-   *     starts at position 0.
+   *                   the start of the item at {@code startIndex}. If {@link C#TIME_UNSET} is passed, playback
+   *                   starts at position 0.
    * @return The Cast {@code PendingResult}, or null if no session is available.
    */
   public PendingResult<MediaChannelResult> loadItem(MediaQueueItem item, long positionMs) {
-    return loadItems(new MediaQueueItem[] {item}, 0, positionMs, REPEAT_MODE_OFF);
+    pendingSeekWindowIndex = 0;
+    currentWindowIndex = 0;
+    return loadItems(new MediaQueueItem[]{item}, 0, positionMs, REPEAT_MODE_OFF);
   }
 
   /**
    * Loads a media queue. If no session is available, does nothing.
    *
-   * @param items The items to load.
+   * @param items      The items to load.
    * @param startIndex The index of the item at which playback should start.
    * @param positionMs The position at which the playback should start in milliseconds relative to
-   *     the start of the item at {@code startIndex}. If {@link C#TIME_UNSET} is passed, playback
-   *     starts at position 0.
+   *                   the start of the item at {@code startIndex}. If {@link C#TIME_UNSET} is passed, playback
+   *                   starts at position 0.
    * @param repeatMode The repeat mode for the created media queue.
    * @return The Cast {@code PendingResult}, or null if no session is available.
    */
   public PendingResult<MediaChannelResult> loadItems(MediaQueueItem[] items, int startIndex,
-      long positionMs, @RepeatMode int repeatMode) {
+                                                     long positionMs, @RepeatMode int repeatMode) {
     if (remoteMediaClient != null) {
       positionMs = positionMs != C.TIME_UNSET ? positionMs : 0;
       waitingForInitialTimeline = true;
```

Comment:
Fix for:
3. ArrayIndexOutOfBoundsException: length=2; index=2 when setting new media sources and some other problems with wron current position.